src/include/mpir_gpu_util.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#ifndef MPIR_GPU_UTIL_H_INCLUDED
+#define MPIR_GPU_UTIL_H_INCLUDED
+
+/* If buf is device memory, allocate a host buffer that can hold the data,
+ * return the host buffer, or NULL if buf is not device memory */
+MPL_STATIC_INLINE_PREFIX void *MPIR_gpu_host_alloc(const void *buf,
+                                                   MPI_Aint count, MPI_Datatype datatype)
+{
+    MPL_pointer_attr_t attr;
+    MPIR_GPU_query_pointer_attr(buf, &attr);
+
+    if (attr.type != MPL_GPU_POINTER_DEV) {
+        return NULL;
+    } else {
+        MPI_Aint extent, true_lb, true_extent;
+        MPIR_Datatype_get_extent_macro(datatype, extent);
+        MPIR_Type_get_true_extent_impl(datatype, &true_lb, &true_extent);
+        // extent = MPL_MAX(extent, true_extent);
```

Comment:
I'm somewhat surprised this works without this line. In any case, if it's not needed, we should remove it.

Reply:
Yeah, I meant to remove it but have forgotten. The data is packed by `extent` not `true_extent`, so I believe take `true_extent * count` is wrong or at least imprecise. `true_extent` should only be used to adjust the final boundary.